### PR TITLE
[Apache Tomcat] Add JDBC Connection Pool maxActive on Ingestion Pipeline // do not approve

### LINKED ISSUE #10037: [Apache Tomcat] Add JDBC Connection Pool maxActive on Ingestion Pipeline
Add the extraction of the connection pool maxActive attribute on the Apache Tomcat ingest pipeline.

JMX field: prometheus.metrics.Catalina_DataSource_maxActive

### PR DESCRIPTION
**Pull request by MatheusGelinskiPires**
## Type

- Enhancement

## Proposed commit message

Extract the maxActive field from the Apache Tomcat connection pool using the ingest pipeline.

This data could be used to identify the usage of a JBDC pool.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Tested in a real case adding this change to the default ingest pipeline.

## How to test this PR locally

1. Install elastic-agent
2. Add the Apache Tomcat integration

## Related issues

- Closes #10037

## Screenshots

None

### CODE REVIEW

**Comment by MatheusGelinskiPires** [comment]
Please label the PR as an enhancement as I have no permissions to set labels.

**Comment by MatheusGelinskiPires** [comment]
> ❌ Author of the following commits did not sign a [Contributor Agreement](https://www.elastic.co/contributor-agreement): [67441af](https://github.com/elastic/integrations/commit/67441aff1647a3b03b683de7329a897f264c0592), [f888953](https://github.com/elastic/integrations/commit/f888953468e028f300b5ded08b106ac1a63f44d7), [48bdfe5](https://github.com/elastic/integrations/commit/48bdfe56da51ef836e1a09a31337a72d7be95adc)
> 
> Please, read and sign the above mentioned agreement if you want to contribute to this project

already done

File: packages/apache_tomcat/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.5.2"
+  changes:
+    - description: Add JDBC Connection Pool maxActive on Ingestion Pipeline.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/10037
```
Comment (aliabbas-elastic) [inline packages/apache_tomcat/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/10038
```

File: packages/apache_tomcat/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.5.2"
```
Comment (aliabbas-elastic) [inline packages/apache_tomcat/changelog.yml:2]: Please update this same version in `packages/apache_tomcat/manifest.yml`

**Review by aliabbas-elastic — CHANGES_REQUESTED** [review]
After doing these changes please run once `elastic-package build` so that the new metric introduced is displayed in `README`

**Comment by aliabbas-elastic** [comment]
Also, as you're introducing a new metric to the data stream, it's better to update the pipeline and system tests

File: packages/apache_tomcat/data_stream/connection_pool/fields/fields.yml [hunk 3]
```diff
@@ -181,6 +181,10 @@
               type: double
               description: Maximum total of connection pool.
               metric_type: gauge
+            - name: active
+              type: double
+              description: Maximum active of connection pool.
```
Comment (shmsr) [inline packages/apache_tomcat/data_stream/connection_pool/fields/fields.yml:186]: The description sounds incomplete. Not just this one but for fields like `total`.

>  The maximum number of active connections that can be allocated from this pool at the same time.

^^ I got it from the doc itselfl.

**Comment by MatheusGelinskiPires** [comment]
> Also, as you're introducing a new metric to the data stream, it's better to update the pipeline and system tests

the maxActive is already on the [ context.xml](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/_dev/deploy/docker/configs/context.xml#L14) file but I didn't found the maxTotal in there.

**Comment by MatheusGelinskiPires** [comment]
> After doing these changes please run once `elastic-package build` so that the new metric introduced is displayed in `README`

done

**Comment by aliabbas-elastic** [comment]
> the maxActive is already on the [context.xml](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/_dev/deploy/docker/configs/context.xml#L14) file but I didn't found the maxTotal in there.

You need to basically add the metric that you are getting in live response [here](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/data_stream/connection_pool/_dev/test/pipeline/test-conection-pool-metrics.json). And then run `elastic-package test pipeline -v -g` which would test the changes done. After that passes, please also run `elastic-package test system -v -d connection_pool -g` which would run the system tests and update the `sample_event.json`

**Comment by MatheusGelinskiPires** [comment]
> > the maxActive is already on the [context.xml](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/_dev/deploy/docker/configs/context.xml#L14) file but I didn't found the maxTotal in there.
> 
> You need to basically add the metric that you are getting in live response [here](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/data_stream/connection_pool/_dev/test/pipeline/test-conection-pool-metrics.json). And then run `elastic-package test pipeline -v -g` which would test the changes done. After that passes, please also run `elastic-package test system -v -d connection_pool -g` which would run the system tests and update the `sample_event.json`

got it!

I will do it and also create another PR to fix the CLA issues.

File: packages/apache_tomcat/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
...
+  changes:
+    - description: Add JDBC Connection Pool maxActive on Ingestion Pipeline.
```
Comment (shmsr) [inline packages/apache_tomcat/changelog.yml:4]:
```suggestion
    - description: Add JDBC Connection Pool's `maxActive` in the ingest pipeline.
```

File: packages/apache_tomcat/changelog.yml (hunk 2 above)
Comment (shmsr) [inline packages/apache_tomcat/changelog.yml:2]: With a patch upgrade, we should not really do field modifications. This is at least a minor upgrade IMO i.e., 1.6.0

**Review by shmsr — CHANGES_REQUESTED** [review]

**Comment by MatheusGelinskiPires** [comment]
Closing this PR because it was replace by the #10069

**Pull request closed without merging**